app/forms/hyrax/forms/dashboard/nest_collection_form.rb

```diff
@@ -104,17 +102,6 @@ def remove
 
         attr_accessor :query_service, :persistence_service, :context, :collection
 
-        # ideally we would love to be able to eliminate collections which exceed the
-        # maximum nesting depth from the lists of available collections, but the queries
-        # needed to make the determination are too expensive to do for every possible
-        # collection, so we only test for this situation prior to saving the new
-        # relationship.
-        def nesting_within_maximum_depth
```

Comment:
i'm trying to write a commit message for the backport of this change---which no-ops this validation if the solr graph query is configured. but i'm a bit stuck and realizing i'm not entirely clear on why we're removing this. can you help me @cjcolvar?

Reply:
I assumed that the max depth was only needed to limit the amount of nested reindexing to undertake.  Now that the indexing isn't necessary the technical limit seems like solr's graph query parser's efficiency.  If there is a functional reason for having a limit (and making it adopter configurable) then I could restore the depth limit enforcement with a new implementation.  The configuration of the max depth was also on the Samvera::NestingIndexer so we'd have to make a new config in Hyrax for it.

Reply:
thanks! super helpful